src/shogun/machine/BaggingMachine.cpp

```diff
@@ -44,17 +47,53 @@ CBaggingMachine::~CBaggingMachine()
 
 CBinaryLabels* CBaggingMachine::apply_binary(CFeatures* data)
 {
-	SGVector<float64_t> combined_vector = apply_get_outputs(data);
+	SGMatrix<float64_t> output = apply_outputs_without_combination(data);
+
+	CMeanRule* mean_rule = new CMeanRule();
```

Comment:
I'm a little confused when to use UNREF. When should it be used? There are some cases where new objects are created, UNREF is not run and there are no memory leaks.